src/runtime/xen-state.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright (c) 2020 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+// tslint:disable:no-any
+
+const nob = () => Object.create(null);
+
+// @ts-ignore
+const debounce = (key, action, delay) => {
```

Comment:
Does this mean we will have now 2 divergent versions of xen-state? If so, is there any way to avoid it? E.g. adding .d.ts file instead of making a full on copy?

Reply:
In the OSS world, there's no way to have a mixture of typescript and javascript in a library. I tried to find a way around this, but this was the only way that seemed trackable (I didn't want to create custom bazel rules to handle this case). So, there wasn't an option to add a `.d.ts` file. :(